### PR TITLE
use rbenv+ruby-build instead of rvm

### PR DESCRIPTION
**Pull request by flavorjones**
While working on https://github.com/rake-compiler/rake-compiler-dock/issues/75 I found that RVM has issues on Alpine, specifically https://github.com/rvm/rvm/issues/5255

Now, while we could probably use another musl-based image (and not Alpine specifically, since the RVM problem is really related to Alpine's use of busybox), I will point out that ruby-build appears to be more actively maintained than rvm, is simpler, and is a bit easier to work with. So I think this is a good change regardless (pending green tests).

Changes:
- user `rvm` no longer exists, replaced by `rubyuser`
- `rvm` has been replaced by `rbenv` and `ruby-build`
  - no longer applying the sendfile patches to Ruby
  - no longer updating gems belonging to the bootstrap rubies

Also in this PR, I've updated the predefined user and group lists to be the superset of all our images.

### CODE REVIEW

**Comment by flavorjones** [comment]
@larskanis are you OK with this change? can you let me know if you have concerns or want to get on a call to discuss it?

**Comment by larskanis** [comment]
Sorry @flavorjones for the delay! In general I'm fine with the change.

The sendfile patches were necessary for VirtualBox. I guess this is still the case, but VirtualBox emulation for docker is not very common these days. So we can remove them.

In any case: Great work!

**Review by larskanis — APPROVED** [review]

File: Dockerfile.mri.erb [hunk 1]
```diff
@@ -89,48 +40,84 @@ RUN dpkg -i /debs/*.deb
...
-        cd /usr/local/rvm/gems/ruby-\$v/gems/rake-compiler-1.2.5 && \
-        echo applying patches to ruby-\$v /home/rvm/patches/rake-compiler-1.2.5/*.patch && \
-        ( git apply /home/rvm/patches/rake-compiler-1.2.5/*.patch || true ) \
+      echo 'gem: --no-ri --no-rdoc --no-document' >> ~/.gemrc && \
+      export CFLAGS='-s -O3 -fno-fast-math -fPIC' && \
+      for v in ${RBENV_RUBIES} ; do \
+        rbenv install \$v -- --disable-install-doc ; \
+      done && \
+      find ${RBENV_ROOT} -type d -print0 | sudo xargs -0 chmod g+w \
+    "
```
[the remark this answers is not in the document]
Comment (flavorjones) [reply Dockerfile.mri.erb]: Of course! Just pushed this change.

[the remark this answers is not in the document]
Comment (larskanis) [reply Dockerfile.mri.erb]: Nice! And happy new year Mike! I have a question to you. We recently watched the movie "Hackers" from 1995. As you probably know, it takes place in New York. I proudly told the audience that "I know one of these guys - he is called @flavorjones and he's hacking Ruby now. He's probably still using these roller skaters to cross the city!" Do you?

**Pull request merged**